lib/jekyll/post.rb

```diff
@@ -118,6 +119,14 @@ def title
       data.fetch('title') { titleized_slug }
     end
 
+    # Public: the Post excerpt_separator, from the YAML Front-Matter or site default
+    #         excerpt_separator value
+    #
+    # Returns the post excerpt_separator
+    def excerpt_separator
+      data.fetch('excerpt_separator') { site.config['excerpt_separator'].to_s }
```

Comment:
`||` is better safe than sorry, people are people and people make mistakes, it's possible.  `Hash#fetch` does not protect against basic cases of human editing, human assumption and human mistakes.  Trust me, it happens, I've done it before.